packages/nitro-protocol/contracts/interfaces/IAssetHolder.sol

```diff
@@ -45,10 +45,9 @@ interface IAssetHolder {
     );
 
     /**
-     * @dev Indicates that `amount` assets have been transferred (internally or externally) to the destination denoted by `destination`.
+     * @dev Indicates that `amounts` assets have been transferred (internally or externally) to the destinationss pecified the CALLDATA. allocation[indices[k]] was transferred amounts[k]
      * @param channelId The channelId of the funds being withdrawn.
-     * @param destination An internal destination (channelId) of external destination (padded ethereum address)
-     * @param amount Number of assets transferred (wei or tokens).
+     * @param amounts Number of assets transferred (e.g. wei or tokens) to the destinations specified in the calldata
      */
-    event AssetTransferred(bytes32 indexed channelId, bytes32 indexed destination, uint256 amount);
+    event AssetsTransferred(bytes32 indexed channelId, uint256[] amounts);
```

Comment:
Do you have an algorithm in mind for re-constructing the destinations in the client?
I suppose the client can get:

    the outcome pre-transfer (calldata)
    the indices transferred (calldata)
    the amounts transferred (calldata)

Now, if this is the data that your algorithm uses, you might as well emit the sum amount transferred here, no? That would save 256 gas per payout.

Reply:
> the outcome pre-transfer (calldata) < YES
the indices transferred (calldata) < YES
the amounts transferred (calldata) < I don't think so?

Yeah the algorithm is:

```
oldAllocation = abi.decode(calldata.allocationBytes);
newAllocation = oldAllocation.clone();
calldata.indices.forEach(i=> newAllocation[i].amount -= event.payouts[i]);
return newAllocation
```

I don't think this is the absolute minimum, though. We could replace `uint256[] payouts` with `uint256 initialHoldings`, and then emulate `_computeNewAllocation` in the client code.

Reply:
> > the amounts transferred (calldata) < I don't think so?

(mistake, meant to be (this event))

Reply:
> [andrew accidentally edited this comment]

I don't remember exactly what you wrote, so I'm going to stop editing it now. but we're on the same page: you can emit either:
1. the total amount transferred
2. the initial holdings
and you can re-construct the individual payouts, and only pay the gas to emit one uint256 instead of an array

(2) is what you suggested, and that probably makes sense?

Reply:
Yeah that's it

Reply:
I took the liberty to implement (2) so we can compare the gas (though I suppose we could predict the gas fee reduction).